promising/promises.py

```
@@ -0,0 +1,541 @@
+        #  should be reconsidered when the library is used by MiniAgents and
+        #  real life scenarios become clear.
+        # TODO Issue a warning if start_soon is False, but make_parent_wait is
+        #  True ? Prohibit such combinations altogether ? I'd say let's
+        #  prohibit it altogether.
+        self._children: WeakSet[Promise[Any]] = WeakSet()
+
+        if self._parent is not None:
+            if loop is None:
+                loop = self._parent._loop
+            elif loop is not self._parent._loop:
+                raise ValueError("Parent and child Promises must share the same event loop")
```

Ruff (0.14.14)

[warning] Avoid specifying long messages outside the exception class

(TRY003)